int64 e0
int64 e1
int16 a0
int16 a1
int16 a2
int16 a3
int16 a4
int16 a5
int32 u0
int32 u1
int32 u2
time t